workflow/quickstarts/fastapi.mdx

```diff
@@ -0,0 +1,281 @@
+        print("step 1 input", input, "output", output)
+        return output
+
+    result1: str = await context.run("step1", _step1)
+
+    response: CallResponse[Dict[str, str]] = await context.call(
+        "get-data",
+        url=f"{context.env.get('UPSTASH_WORKFLOW_URL', 'http://localhost:8000')}/get-data",
+        method="POST",
+        body={"message": result1},
+    )
+
+    async def _step2() -> str:
+        output = some_work(response.body["message"])
+        print("step 2 input", response, "output", output)
+        return output
+
+    await context.run("step2", _step2)
+
+```
+</Tab>
+<Tab title="Auth">
+```python main.py
+from fastapi import FastAPI
+from upstash_workflow.fastapi import Serve
+from upstash_workflow import AsyncWorkflowContext
+
+app = FastAPI()
+serve = Serve(app)
+
+
+def some_work(input: str) -> str:
+    return f"processed '{input}'"
+
+
+@serve.post("/auth")
+async def auth(context: AsyncWorkflowContext[str]) -> None:
+    if context.headers.get("authentication") != "Bearer secret_password":
+        print("Authentication failed.")
+        return
+
+    async def _step1() -> str:
+        return "output 1"
+
+    await context.run("step1", _step1)
+
+    async def _step2() -> str:
+        return "output 2"
+
+    await context.run("step2", _step2)
+
+```
+</Tab>
+</Tabs>
+
+## Step 4: Run the Workflow Endpoint
+
+Upstash Workflow is powered by [QStash](/qstash/overall/getstarted), and QStash needs a publicly accessible URL to run your workflows. Here's how to [set up your workflow endpoint for local development](/workflow/howto/local-development).
+
+Once you have a live URL, proceeed with either one of the following steps:
+
+### Setting an enviroment variable (recommended)
+
+Set your publically accessible URL as an environment variable. This variable is only needed for local development and doesn't need to be set in production:
+
+```bash .env.local
+export UPSTASH_WORKFLOW_URL=https://<YOUR_NGROK_OR_TUNNEL_URL>/
+```
+
+### Using the `base_url` option
+
+As an alternative to setting an environment variable, you can also use the `base_url` option in the `@serve.post` decorator. This option is only needed for local development and can be omitted in production:
+
+```python main.py
+@serve.post("/api/example", base_url="https://<YOUR_NGROK_OR_TUNNEL_URL>/")
+async def example(context: AsyncWorkflowContext[str]) -> None: ...
+
+```
+
+### Triggering the workflow
+
+Don't forget to source your environment file to set your environment variables:
+
+```bash Terminal
+source .env.local
+```
+
+After setting your live URL as the environment variable or `base_url` option, trigger your workflow by first starting your FastAPI app:
+
+```bash Terminal
+uvicorn main:app --reload
+```
+
+and then making a POST request to your workflow endpoint. For each workflow run, a unique workflow run ID is returned:
+
+```bash Terminal
+curl -X POST https://<YOUR_NGROK_OR_TUNNEL_URL>/api/workflow
```

Comment:
again, in all examples, can you replace NGROK URL with localhost?

localhost will work in both cases: NGROK or local QStash server